core/app/models/spree/credit_card.rb

```
@@ -42,20 +57,22 @@ class CreditCard < Spree::Base
 
     before_validation :set_default_cc_type
 
+    # Set the expiry month and year
+    #
+    # @param expiry [#to_s]
+    #
+    # @return [undefined]
+    #
+    # @api private
     def expiry=(expiry)
-      return unless expiry.present?
-
-      self.month, self.year =
-        if expiry.match(/\d{2}\s?\/\s?\d{2,4}/) # will match mm/yy and mm / yyyy
-          expiry.delete(' ').split('/')
-        elsif match = expiry.match(/(\d{2})(\d{2,4})/) # will match mmyy and mmyyyy
-          [match[1], match[2]]
-        end
-      if year
-        self.year = '20' + year if year.length == 2
-        self.year = year.to_i
+      match = expiry.to_s.match(EXPIRY_PATTERN)
+      if match
+        year = YEAR_PREFIX + (Integer(match[:year]) % YEAR_DIVISOR)
+        date = Date.new(year, match[:month].to_i)
```

As the `:month` named capture AFAIK is not optional we likely do want to use `Integer` here? `nil.to_i` would silently coerce to `0`?